### PR TITLE
`FixCurrents` objective

### PR DESCRIPTION
**Pull request by ddudt**
New linear objective to fix the currents in a `Coil` or `CoilSet`. 

TO-DO:

- [ ] Add another linear objective to constraint the sum of coil currents within a coil set? This could be a separate PR
- [x] Tests? See comment below, this will be covered in #977

### CODE REVIEW

**Comment by f0uriest** [comment]
Maybe `FixCoilCurrent(s)` to make it more explicit and avoid confusion?

**Comment by ddudt** [comment]
Similar to #1016, I already have plans to use this in coil optimization tests for #977. In an effort to not add redundant unit tests, I vote to skip testing on this PR and wait to cover the new objective in that PR.

**Review by dpanici — CHANGES_REQUESTED** [review]
I think there still should be tests here as it is not clear to me if this can be used to fix only the currents in a subset of coils or to fix every current inside the given coil or coilset (which I think the latter is true). This does not allow the ability to, say, forgo adding the toroidal flux constraint for a vacuum coil optimization by fixing the current in a single coil and allowing the rest to vary

**Comment by ddudt** [comment]
> I think there still should be tests here as it is not clear to me if this can be used to fix only the currents in a subset of coils or to fix every current inside the given coil or coilset (which I think the latter is true). This does not allow the ability to, say, forgo adding the toroidal flux constraint for a vacuum coil optimization by fixing the current in a single coil and allowing the rest to vary

Before it was always fixing all of the currents in a `CoilSet`. Now I added an `indices` option to specify fixing only a subset of the coil currents. Also added a basic test. If we want to get fancy, we could wait until PR #995 is merged and add a test optimizing for the coil currents in some coils with others fixed. But this will still also get used in an optimization test in #977.

File: desc/objectives/linear_objectives.py
Comment (dpanici) [inline desc/objectives/linear_objectives.py:2637]: could this instead be more like the optimizable way where it is a pytree? Or could it be both? to allow for fixing currents when things are nested

File: desc/objectives/linear_objectives.py
Comment (f0uriest) [inline desc/objectives/linear_objectives.py:2623]: wrong default

File: tests/test_linear_objectives.py
Comment (f0uriest) [inline tests/test_linear_objectives.py:183]: personally I prefer the explicit `desc.examples.get`, since `get` by itself is pretty generic, and there are other methods in python with the same name

File: desc/objectives/linear_objectives.py
Comment (f0uriest) [inline desc/objectives/linear_objectives.py:2637]: I think it would be good to include an example of this (can basically just use the one from the test). Also the part about `Must have the same pytree structure as coil.params_dict.` isn't quite right I think, because its a list of bools, not a list of dict of bool? Not sure the best way to describe that, so I think an example would be clearest.

**Review by f0uriest — CHANGES_REQUESTED** [review]

File: desc/objectives/linear_objectives.py
Comment (dpanici) [inline desc/objectives/linear_objectives.py:2624]: does the target also need the same structure as the indices?

File: desc/objectives/linear_objectives.py
Comment (dpanici) [inline desc/objectives/linear_objectives.py:2624]: going to be annoying again, so this should not be a bool right? it should be either the desired current, or None in which case it will default to the coils current?

Comment (ddudt) [reply desc/objectives/linear_objectives.py]: Correct, thanks for catching this. `target` gets passed through to `FixParameters` so it should be the same pytree structure as `coil.params_dict`. I copied the docstring from that parent class

**Review by f0uriest — APPROVED** [review]

**Review by dpanici — APPROVED** [review]

**Pull request merged**
